neusomatic/python/call.py

```diff
@@ -104,181 +98,190 @@ def call_variants(net, vartype_classes, call_loader, out_dir, model_tag, use_cud
 
 
 def pred_vcf_records_path((path, true_path_, pred_all, chroms, vartype_classes, ref_file)):
-    fasta_file = pysam.FastaFile(ref_file)
-    ACGT = "ACGT"
-    I = imread(true_path_) / 255.0
-    vcf_record = None
-    Ih, Iw, _ = I.shape
-    zref_pos = np.where((np.argmax(I[:, :, 0], 0) == 0) & (
-        sum(I[:, :, 0], 0) > 0))[0]
-    nzref_pos = np.where(
-        (np.argmax(I[:, :, 0], 0) > 0) & (sum(I[:, :, 0], 0) > 0))[0]
-    # zref_pos_0 = np.where((I[0, :, 0] > 0) & (sum(I[:, :, 0], 0) > 0))[0]
-    # nzref_pos_0 = np.where((I[0, :, 0] == 0) & (sum(I[:, :, 0], 0) > 0))[0]
-    # assert(len(set(zref_pos_0)^set(zref_pos))==0)
-    # assert(len(set(nzref_pos_0)^set(nzref_pos))==0)
-
-    chrom, pos, ref, alt, _, center, _, _, _ = path.split(
-        ".")
-    center = int(center)
-    pos = int(pos)
-
-    ins_no_zref_pos = False
-    infered_type_pred = "NONE"
-    pred = list(pred_all)
-    min_acceptable_probmax = 0.05
-    center_dist_roundback = 0.8
-    too_far_center = 3
-    neigh_bases = 7
-    while True:
-        pred_probs = pred[3]
-        if sum(pred_probs) < min_acceptable_probmax:
-            break
-        amx_prob = np.argmax(pred_probs)
-        type_pred = vartype_classes[amx_prob]
-        if type_pred == "NONE":
-            break
-        center_pred = min(max(0, pred[1][0]), Iw - 1)
-        if abs(center_pred - center) < center_dist_roundback:
-            center_ = center
-        else:
-            center_ = int(round(center_pred))
-        if type_pred != "INS" and center_ in zref_pos:
-            if center in nzref_pos:
-                center_ = center
-            else:
-                center_ = nzref_pos[
-                    np.argmin(abs(nzref_pos - center_pred))]
-        elif type_pred == "INS" and center_ in nzref_pos:
-            if len(zref_pos) == 0:
-                if len(alt) < len(ref) + neigh_bases:
-                    break
+    thread_logger = logging.getLogger(
+        "{} ({})".format(pred_vcf_records_path.__name__, multiprocessing.current_process().name))
+    try:
+        fasta_file = pysam.FastaFile(ref_file)
+        ACGT = "ACGT"
+        I = imread(true_path_) / 255.0
+        vcf_record = []
+        Ih, Iw, _ = I.shape
+        zref_pos = np.where((np.argmax(I[:, :, 0], 0) == 0) & (
+            sum(I[:, :, 0], 0) > 0))[0]
+        nzref_pos = np.where(
+            (np.argmax(I[:, :, 0], 0) > 0) & (sum(I[:, :, 0], 0) > 0))[0]
+        # zref_pos_0 = np.where((I[0, :, 0] > 0) & (sum(I[:, :, 0], 0) > 0))[0]
```

Comment:
Do we need these commented out lines? If no, then please remove.

Reply:
There are some stuff from before I want to check later. I have commented for now but needs to get back to them once have some time.